cli/flox-activations/src/activations.rs

```diff
@@ -122,8 +128,8 @@ pub struct Activation {
 }
 
 impl Activation {
-    pub fn id(&self) -> Uuid {
-        self.id
+    pub fn id(&self) -> String {
+        self.id.clone()
     }
```

Comment:
suggestion: since in most cases we seem to just read the id a=t compare it, we don't need to create an owned value here

```suggestion
    pub fn id(&self) -> &str {
        self.id
    }
```

Reply:
I think when I tried that I ran into
```
error[E0502]: cannot borrow `activations` as immutable because it is also borrowed as mutable
   --> flox-activations/src/activations.rs:335:26
    |
332 |         let activation = activations.create_activation(store_path, 123).unwrap();
    |                          ----------- mutable borrow occurs here
...
335 |         let activation = activations.activation_for_id_ref(&id).unwrap();
    |                          ^^^^^^^^^^^                       --- mutable borrow later used here
    |                          |
    |                          immutable borrow occurs here
```

Happy to approve something but I didn't feel like fighting the borrow checker